web/src/emoji.ts

```diff
@@ -1,34 +1,93 @@
 import _ from "lodash";
 
+type RealmEmoji = Record<
+    string,
+    {
+        id: number;
+        author_id: number;
+        deactivated: boolean;
+        name: string;
+        source_url: string;
+        still_url: string | null;
+    }
+>;
+
+type EmojiCodes = {
+    codepoint_to_name: Record<string, string>;
+    name_to_codepoint: Record<string, string>;
+    emoji_catalog: Record<string, string[]>;
+    emoticon_conversions: Record<string, string>;
+    names: string[];
+};
+
+type EmojiParams = {
+    realm_emoji: RealmEmoji;
+    emoji_codes: EmojiCodes;
+};
+
+type Translation = {
```

Comment:
Let's call this `EmoticonTranslation`, since this is specifically for emoticons.